scripts/finish-release.ts

```diff
@@ -0,0 +1,210 @@
+  git.merge(`${upstreamName}/${releaseBranch}`, { message: 'chore: merge release commit from main to dev' });
+}
+type Args = {
+  continue: boolean;
+};
+function getArgs(): Args {
+  const args = process.argv.slice(2);
+  return {
+    continue: args.length > 0 && args.indexOf('--continue') > -1,
+  };
+}
+
+process.on('uncaughtException', (err) => {
+  if (err.message === 'dirty working tree') {
+    process.exit(1);
+  }
+});
+
+export async function main() {
+  const args = getArgs();
+  const git = new Git();
+  const rlInterface = rl.createInterface({
+    input: stdin,
+    output: stdout,
+  });
+
+  if (!git.isCleanWorkingTree()) {
+    console.error('Please run the script from a clean working tree');
+    process.exit(2);
+  }
+
+  const releaseSha = git.getShortSha('main');
+  const DEV_BRANCH = 'dev';
+  const RELEASE_BRANCH = 'main';
+  const REPO_NAME = 'aws-amplify/amplify-cli';
+  const mergeBranch = 'dev-main-merge-' + releaseSha;
+  const remoteOutput = git.remote(true);
+  const upstreamName = extractUpstreamNameFromRemotes(remoteOutput, REPO_NAME);
+
+  const doesMergeBranchExist = git.isExistingBranch(mergeBranch);
+
+  if (!upstreamName) {
+    console.error('could not find remote name for the aws-amplify/amplify-cli repository');
+    process.exit(1);
+  }
+  if (!args.continue && doesMergeBranchExist) {
+    console.error(
+      `The merge branch \`${mergeBranch}\` already exists. Please delete the merge branch and try again, or run the script with \`--continue\``,
+    );
+    process.exit(1);
+  }
+  const canContinue = args.continue || (await shouldContinue(rlInterface, `This will create the merge branch \`${mergeBranch}\`.`));
+  if (!canContinue) {
+    console.info('Exiting...');
+    process.exit(0);
+  }
+
+  console.info('Switching to branch ' + DEV_BRANCH);
+  git.checkout(DEV_BRANCH);
+  git.pull();
+  console.info('Creating merge branch ' + mergeBranch);
+  try {
+    const shouldAttemptCreate = !doesMergeBranchExist;
+    git.checkout(mergeBranch, shouldAttemptCreate);
+  } catch (e) {
+    console.error('Failed to create merge branch, exiting...');
+    process.exit(1);
+  }
+  console.info(`Fetching ${upstreamName}/${RELEASE_BRANCH}`);
+  git.fetch(upstreamName, RELEASE_BRANCH);
+
+  console.info(`Merging ${upstreamName}/${RELEASE_BRANCH} into ${mergeBranch}`);
+  git.merge(`${upstreamName}/${RELEASE_BRANCH}`, { message: 'chore: merge release commit from main to dev' });
+
+  if (!git.isCleanWorkingTree()) {
+    console.info('Resolve merge conflicts and commit merge, then run `yarn run finish-release --continue`');
+    process.exit(0);
+  }
+
+  git.push(upstreamName, mergeBranch);
+
+  console.info(`
+You can compare changes between the branches here: ${getCompareLink(
+    DEV_BRANCH,
+    mergeBranch,
+  )}. To finish the release, open a pull request between ${mergeBranch} into ${DEV_BRANCH}
+`);
+
+  process.exit(0);
+}
+
+if (require.main === module) {
+  main().catch(() => process.exit(1));
```

Comment:
could we try this https://stackoverflow.com/questions/46515764/how-can-i-use-async-await-at-the-top-level ?

Reply:
my preferred way to do this is:
```
const main = async () => { /* stuff */ }

main().catch(console.error)
```

Reply:
I have to add the `require.main == module` statement in there so I can import the file for unit tests. Otherwise the script executes when importing for tests.